### PR TITLE
Add developer API to mirror triggerHandler behavior

### PR DESCRIPTION
**Pull request by jondavis9898**
See Issue #196 and PR #208 for background

### CODE REVIEW

File: rolluptool/src/classes/RollupServiceTest3.cls [hunk 1]
```diff
@@ -528,6 +528,539 @@ private with sharing class RollupServiceTest3
...
+
+		// combine the new and old
+		List<SObject> newAndOldChildren = children.clone();
+		newAndOldChildren.addAll(newChildren);
+
+		// Call developer 'trigger like' API simulating an update
+		RollupService.rollup(new Map<Id, SObject>(children), new Map<Id, SObject>(newAndOldChildren), childType);
+
+		// Assert parents are updated
+		assertParents = new Map<Id, SObject>(Database.query(String.format('select id, {0} from {1}', new List<String>{ aggregateResultField, parentObjectName })));
```
Comment (afawcett) [inline rolluptool/src/classes/RollupServiceTest3.cls]: Might be worth checking Limits class before and after have not changed (within expected tolerances) for SOQL and DML usage for super paranoid testing? ;-)

**Comment by jondavis9898** [comment]
Yeah, I covered some Limits assertions in one of the tests but I can definitely add some more.  Will throw some in and update PR.  Everything else look OK and inline with what you were thinking?

**Comment by afawcett** [comment]
Yes for sure, loving it actually, the tests are soooo important for this thing, its gather quite a lot of users, using it in mission critical situations, a regression would not be good! ;-)

**Comment by jondavis9898** [comment]
agreed and yes the library is awesome and seems to have a gained a lot of momentum.  Fortunately the changes to existing code path are rather minor but still want to make sure we have all bases covered.

**Comment by afawcett** [comment]
Cool, let me know when its ready to merge.

**Comment by jondavis9898** [comment]
Hi @afawcett - In adding in the limit assertions to these tests, I was encountering some unexpected results and have opened #226 and #227.  Let me know your thoughts on those and once decided upon, I'll wrap up this PR accordingly.  Thanks!

**Pull request merged**